solved the task

ogerenko opened the pull request:
- [DEMO LINK](https://ogerenko.github.io/layout_product-cards/)
- [TEST REPORT LINK](https://ogerenko.github.io/layout_product-cards/report/html_report/)

- [x] there 2 BEM blocks `card` and `stars` each in it own file
- [x] SCSS Nesting is used for `elements`, `modifiers` and `pseudo-classes`
- [x] SCSS Variables are used for main values and placed in a **separate file**
- [x] all `stars--N` modifiers work as expected (Highlight first `N` stars)
- [x] Code follows all the [Code Style Rules ❗️](https://mate-academy.github.io/layout_task-guideline/html-css-code-style-rules)

ogerenko commented:
I don’t know if this is a mistake on your part or if I didn’t read the layout correctly, but I had problems with the tests when I set the margin between the image and the product name.
the layout says 40 but in order for the tests to pass I had to set it to 36

sTorba24 commented on src/styles/blocks.scss/card-info.scss:
```diff
@@ -0,0 +1,52 @@
+
+  &-product-code {
+    font-size: $small-font-size;
+    font-weight: 400;
```
it's default value, can be removed

sTorba24 commented on src/index.html:
```diff
@@ -6,8 +7,68 @@
+  <body class="page">
+    <div class="section page__card" data-qa="card">
+
+      <section class="section__image">
+        <img
+          class="section-image"
+          src="./images/imac.jpeg"
+          alt="APPLE A1419 iMac 27"
+        >
+      </section>
```
you don't really need this section wrapper
remove it, set all paddings for card
here u have outside space near all elements, so just use paddings on card to squeeze content inside such a box
![image](https://github.com/mate-academy/layout_product-cards/assets/55200530/fcb89b2a-8665-4c18-84ba-43e855ee9cf6)


it is the reason why 40px failed, because img is 134px height, section + 4 bottom. So +40 in result we have 44 px between image and text. If you set height directly 134 to section it'll be ready to use 40px margin.
![image](https://github.com/mate-academy/layout_product-cards/assets/55200530/3c5930cd-16bf-48b3-9058-c56dda8313cf)

sTorba24 commented on src/index.html:
```diff
@@ -6,8 +7,68 @@
-    <h1>Product cards</h1>
+  <!-- #endregion -->
+  <body class="page">
+    <div class="section page__card" data-qa="card">
```
```suggestion
    <div class="card" data-qa="card">
```

sTorba24 replied:
could be just so

sTorba24 commented on src/index.html:
```diff
@@ -6,8 +7,68 @@
+    <div class="section page__card" data-qa="card">
+
+      <section class="section__image">
+        <img
+          class="section-image"
+          src="./images/imac.jpeg"
+          alt="APPLE A1419 iMac 27"
+        >
```
```suggestion
        <img
          class="card__image"
          src="./images/imac.jpeg"
          alt="APPLE A1419 iMac 27"
        >
```

sTorba24 replied:
In BEM we name elements like {block}__{element}

sTorba24 replied:
fix everywhere

sTorba24 commented on src/index.html:
```diff
@@ -6,8 +7,68 @@
+        >
+      </section>
+
+      <section class="section__card-info">
```
```suggestion
      <section class="card__info">
```

sTorba24 replied:
but you don't really need it

sTorba24 commented on src/index.html:
```diff
@@ -6,8 +7,68 @@
+
+      <section class="section__card-info">
+
+        <div class="card-info-title">
```
```suggestion
        <div class="card__title">
```

sTorba24 commented on src/index.html:
```diff
@@ -6,8 +7,68 @@
+
+        <div class="card-info-rates">
+
+          <div class="stars stars--4 card-info-rates--stars">
+            <div class="stars__star"></div>
+            <div class="stars__star"></div>
+            <div class="stars__star"></div>
+            <div class="stars__star"></div>
+            <div class="stars__star"></div>
+          </div>
```
```suggestion
          <div class="stars stars--4">
            <div class="stars__star"></div>
            <div class="stars__star"></div>
            <div class="stars__star"></div>
            <div class="stars__star"></div>
            <div class="stars__star"></div>
          </div>
```

sTorba24 replied:
in BEM you can use block inside block, it's ok

sTorba24 commented on src/styles/blocks.scss/card-info.scss:
```diff
@@ -0,0 +1,52 @@
+
+  &-reviews {
+    font-size: $small-font-size;
+    font-weight: 400;
+
+    // line-height: 14px;
```
remove

sTorba24 commented on src/styles/blocks.scss/stars.scss:
```diff
@@ -0,0 +1,17 @@
+  margin: 0 2px;
+}
+
+.stars--4 > .stars__star:nth-child(-n+4) {
+  background-image: url(../images/star-active.svg);
+}
```
implement stars for all modifiers, for 1,2,3,4,5 active stars(copy from `stars` task)

sTorba24 commented on src/styles/blocks.scss/stars.scss:
```diff
@@ -0,0 +1,17 @@
+  background-image: url(../images/star-active.svg);
+}
+
+.stars--4 .stars__star:first-child {
```
```suggestion
.stars .stars__star:first-child {
```

sTorba24 reviewed (changes requested):
Good work! Some suggestions to improve your work and answer to your question

ogerenko commented:
Thank you for your remarks! 
All clear and understandable for me now. 
Now I understand more clearly how to write classes and wrap tags and classes.

etojeDenys commented on src/index.html:
```diff
@@ -6,8 +7,63 @@
+
+      <div class="card__price">
+
+        <div class="card__price-text">
+          Price:
+        </div>
+
+        <div class="card__price-amount">
+          $2,199
+        </div>
+
```
better to use some semantic tags for the text

anastasiiavorobiova commented on src/index.html:
```diff
@@ -6,8 +7,63 @@
+        alt="APPLE A1419 iMac 27"
+      >
+
+      <div class="card__title">
```
```suggestion
      <h3 class="card__title">
```

anastasiiavorobiova commented on src/index.html:
```diff
@@ -6,8 +7,63 @@
+      <div class="card__title">
+        APPLE A1419 iMac 27" Retina
+        5K Monoblock (MNED2UA/A)
+      </div>
```
```suggestion
      </h3>
```

anastasiiavorobiova commented on src/index.html:
```diff
@@ -6,8 +7,63 @@
+
+      <div class="card__button">
+        <a href="#" class="button-link">
+          <button class="button" data-qa="hover">BUY</button>
```
```suggestion
          BUY
```

anastasiiavorobiova commented on src/styles/blocks.scss/button.scss:
```diff
@@ -0,0 +1,20 @@
+  border: 1px solid transparent;
+  border-radius: 5px;
+  background-color: $main-button-color;
+  color: #fff;
```
Consider using variables for repeating colors

anastasiiavorobiova reviewed (changes requested):
Great work! Almost done!

ogerenko commented:
I fixed the semantic tag for the title 
![Снимок экрана 2024-02-12 000329](https://github.com/mate-academy/layout_product-cards/assets/157063432/8c11e083-9d18-4648-9c61-bcf785f7310e)

but I didn't understand why I was supposed to remove the tag button. 
And leave only tag <a> or what? 
I was trying to do it but everything was failing, all styles for button, whatever I changed the classes in styles.

So please explain to me more clearly why and how I am supposed to do that. 

![Снимок экрана 2024-02-12 000338](https://github.com/mate-academy/layout_product-cards/assets/157063432/1875e41c-46aa-48c0-a9a0-b17e27a57d33)

The same thing with variables. I didn't understand your comment. 
What exactly am I supposed to consider?
I set the variable for the main color of the button or I shouldn't do this?

etojeDenys reviewed (approved):
lgtm
